### PR TITLE
added Longhorn 1.4.0

### PR DESCRIPTION
**Pull request by weizhe0422**
#### Pull Request Checklist ####

- [x] Change does not remove any existing Images or Tags in the images-list file
- [x] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [ ] If updating an existing entry, verify the `SOURCE` is still accurate and upstream hasn't been migrated to a new regitry or repo (if they've migrated, a new repo request to EIO is needed to comply with the `SOURCE DESTINATION TAG` pattern)
- [x] New entries are in format `SOURCE DESTINATION TAG`
- [x] New entries are added to the correct section of the list (sorted lexicographically)
- [x] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [x] New entries are licensed with Rancher favored licenses - Apache 2 and MIT - or approved licenses - as according to [CNCF approved licenses](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md).
- [ ] Changes to scripting or CI config have been tested to the best of your ability

#### Additional Notes ####

 - EIO repo creation request: https://github.com/rancherlabs/eio/issues/1426

#### Final Checks after the PR is merged ####
- [ ] Confirm that you can pull the new images and tags from DockerHub

cc @innobead @PhanLe1010 @khushboo-rancher

### CODE REVIEW

**Comment by brandond** [comment]
It looks like this mirrors a couple new images. Can you link to the EIO repo creation issue so that we know when it's closed and can approve?

**Comment by PhanLe1010** [comment]
Note: this one need to wait for Longhorn 1.4.0 to release on Dec 29th

**Comment by weizhe0422** [comment]
@brandond I've created an issue (link: https://github.com/rancherlabs/eio/issues/1426)  to propose adding two new images to the repository including `rancher/mirrored-longhornio-livenessprobe` and `rancher/mirrored-longhornio-support-bundle`.

**Comment by weizhe0422** [comment]
Update: New two image repositories have been created on Docker hub.

ref. https://github.com/rancherlabs/eio/issues/1426

**Review by brandond — APPROVED** [review]

File: images-list
Comment (PhanLe1010) [inline images-list:628]: Support bundle kit is updated to newer version https://github.com/longhorn/longhorn/blob/f48a8ee582691da148a02157ca653a093e92552b/chart/values.yaml#L46

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]

**Comment by PhanLe1010** [comment]
So we are no longer try to keep fix version for instance manager, share manager, backing image manager but tag a new version for every release? i.e.,

```
longhornio/longhorn-instance-manager rancher/mirrored-longhornio-longhorn-instance-manager v1.4.0
longhornio/longhorn-share-manager rancher/mirrored-longhornio-longhorn-share-manager v1.4.0
longhornio/backing-image-manager rancher/mirrored-longhornio-backing-image-manager v1.4.0
```

Do you know the reason for the change? @innobead @shuo-wu ?

**Comment by innobead** [comment]
> So we are no longer try to keep fix version for instance manager, share manager, backing image manager but tag a new version for every release? i.e.,
> 
> ```
> longhornio/longhorn-instance-manager rancher/mirrored-longhornio-longhorn-instance-manager v1.4.0
> longhornio/longhorn-share-manager rancher/mirrored-longhornio-longhorn-share-manager v1.4.0
> longhornio/backing-image-manager rancher/mirrored-longhornio-backing-image-manager v1.4.0
> ```
> 
> Do you know the reason for the change? @innobead @shuo-wu ?

Yes, and this change starts from 1.4. For the previous branches, we will keep the same. Also, the internal version variables to determine the compatibility between client and server is the same as usual. 

https://github.com/longhorn/longhorn/issues/5024

File: images-list
Comment (weizhe0422) [reply images-list]: Thanks for reminding. I've updated the `support-bundle-kit` version as `v0.0.17`.

**Review by brandond — APPROVED** [review]

**Comment by PhanLe1010** [comment]
> Yes, and this change starts from 1.4. For the previous branches, we will keep the same. Also, the internal version variables to determine the compatibility between client and server is the same as usual.

But this approach means that we will deploy new instance manager pods for every Longhorn new release even if there is nothing change in the instance manager. This will create unnecessary work to move the engine/replica processes to the new instance manager pods. Have we considered this disadvantage? 

cc @innobead  @shuo-wu

**Comment by shuo-wu** [comment]
@PhanLe1010 Last time I discussed this issue with David, he explained one main reason for this: We are using the BCI image now, which typically contains new updates for CVEs during each release. Hence updating the instance manager image is indispensable no matter if there is any Longhorn code change

**Comment by PhanLe1010** [comment]
I see. Thanks @shuo-wu

**Comment by innobead** [comment]
The other important thing we should think further is about how to reduce the resource consumption during the live volume migration, because instance manager is already not just a engine/replica process manager only but also becomes the proxy between control plane and data plane. It's heavily coupled with engine now.

**Comment by PhanLe1010** [comment]
@brandond 

Could we please merge this PR? 
There are other works that are being blocked by this PR. 

Thank you

**Comment by brandond** [comment]
I'm not the only one with merge access to this repo, but sure, I can do it.

**Comment by PhanLe1010** [comment]
Thank you very much. Do you know who else we can ping for reviewing/merging?

**Comment by brandond** [comment]
Anyone listed in the CODEOWNERS file:
https://github.com/rancher/image-mirror/blob/4a1699ef6753ad1baa75d2ff1a2be4d6881f3379/CODEOWNERS#L1

**Pull request merged**
